src/lib/ext-activitylog.js

```diff
@@ -0,0 +1,160 @@
+      'show-log-detail-wrapper'
+    )[0];
+    this.showLogDetails = document.getElementById('showLogDetails');
+    this.showLoadedLog = document.getElementById('showLoadedLog');
+    this.table = document.getElementsByTagName('table')[0];
+    this.tableFirstRow = document.querySelector('tbody tr:first-child');
+    this.notice = document.getElementsByClassName('notice')[0];
+  }
+
+  logs = [];
+  loadedLogs = [];
+
+  async getExistingLogs() {
+    const { existingLogs } = await browser.runtime.sendMessage({
+      requestType: 'sendAllLogs',
+      requestTo: 'ext-monitor',
+    });
+    return existingLogs;
+  }
+
+  async loadFile(file) {
+    try {
+      const loadedLogsAsText = await load.loadLogAsText(file);
+      const loadedJSONLogs = JSON.parse(loadedLogsAsText);
+      this.loadedLogs.push(...loadedJSONLogs);
+      this.showLoadedLog.textContent = loadedLogsAsText;
+
+      if (this.notice.textContent.trim().length) {
+        this.notice.textContent = '';
+        this.notice.classList.remove('failure');
+      }
+    } catch (error) {
+      this.notice.textContent = error.message;
+      this.notice.classList.add('failure');
+    }
+  }
+
+  async saveLogs(logs) {
+    try {
+      await save.saveAsJSON(logs);
+
+      if (this.notice.textContent.trim().length) {
+        this.notice.textContent = '';
+        this.notice.classList.remove('failure');
+      }
+    } catch (error) {
+      this.notice.textContent = error.message;
+      this.notice.classList.add('failure');
+    }
+  }
+
+  renderSideBar(logIndex) {
+    const logStringfy = JSON.stringify(this.logs[logIndex]);
+    this.showLogDetails.textContent = logStringfy;
+    this.showLogDetailWrapper.style.display = 'block';
+  }
+
+  messageHandlers = {
+    appendLogs: ({ log }) => {
+      this.logs.push(log);
+      this.renderTable([log]);
+    },
+  };
+
+  handleEvent(event) {
+    if (event.type === 'click') {
+      let logIndex;
+      switch (event.target.tagName) {
+        case 'TD':
+          logIndex = event.target.parentElement.attributes['index'].value - 1;
+          this.renderSideBar(logIndex);
+          break;
+        case 'TR':
+          logIndex = event.target.attributes['index'].value - 1;
+          this.renderSideBar(logIndex);
+          break;
+        case 'default':
+          throw new Error(
+            'wrong event tag name found' + JSON.stringify(event.target.tagName)
+          );
+      }
+    } else {
+      throw new Error('wrong event type found ' + event.type);
+    }
+  }
+
+  renderTable(logs) {
+    for (const log of logs) {
+      const newRow = this.table.insertRow(1);
+      newRow.insertCell(0).textContent = log.id;
+      newRow.insertCell(1).textContent = log.timeStamp;
+      newRow.insertCell(2).textContent = log.type;
+      newRow.insertCell(3).textContent = log.name;
+      newRow.insertCell(4).textContent = log.viewType || 'undefined';
+      newRow.setAttribute('index', this.logs.length);
```

Comment:
To make this less complicated, I am not using `rowIndex` property of table's row. Instead, I am adding an attribute. Since we are using `Array.push` to store logs and table rows are being generated in the opposite way (the latest rows stays at top). I can make the adjustment using `Array.unsift` and use `rowIndex`, if you say. :-)